fog/view/server/src/fog_view_router_server.rs

```diff
@@ -108,11 +107,25 @@ where
             FogViewRouterAdminService::new(shards.clone(), logger.clone()),
         );
         log::debug!(logger, "Constructed Fog View Router Admin GRPC Service");
-
-        // Health check service
-        let health_service =
-            mc_util_grpc::HealthService::new(Some(readiness_indicator.into()), logger.clone())
-                .into_service();
+        let shard_health_clients = shards
+            .read()
+            .expect("RwLock poisoned")
+            .iter()
+            .enumerate()
+            .map(|(i, shard)| {
+                let uri = &shard.uri;
+                log::info!(logger, "HealthClient {i} uri is: {uri}");
+                let channel = ChannelBuilder::default_channel_builder(env.clone())
+                    .connect_to_uri(&shard.uri, &logger);
+                HealthClient::new(channel)
+            })
+            .collect::<Vec<_>>();
+
+        let health_service = mc_util_grpc::HealthService::new(
+            Some(get_router_callback(shard_health_clients, logger.clone())),
+            logger.clone(),
+        )
+        .into_service();
```

Comment:
It's not currently possible for this to work with dynamic shards. You'll need to make a "shards" cache that's shared between the router service and the health service.

Aside: I'm a little confused about `RouterClientListenUri` below, can we chat about it tomorrow?